### PR TITLE
Exceptions thrown during writing should close the connection

### PR DESCRIPTION
**Pull request by moozzyk**
We need to close the connection if there is an exception when writing to the transport on the server side. Currently if an exception happens it leaves the connection in an unsable state - after the exception no messages from the server will be sent to the client because the writing loop is terminated. Ignoring the message could cause hangs on the client side since we can fail while writing a completion message. In this case if the client is awaiting the invocation it will hang because the task will never be completed.

Fixes: #861 and #916

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs [hunk 1]
```diff
@@ -22,6 +22,7 @@
 using Newtonsoft.Json.Serialization;
 using Xunit;
 using Microsoft.AspNetCore.Hosting;
+using System.Runtime.Serialization;
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs]: nit: Sort usings

File: src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs [hunk 2]
```diff
@@ -76,20 +76,27 @@ public class HubEndPoint<THub> : IInvocationBinder where THub : Hub
...
+                                {
+                                    break;
+                                }
                             }
                         }
                     }
                 }
+                catch (Exception ex)
+                {
+                    connectionContext.Abort(ex);
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs]: Came in handy 😄

File: test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs (hunk 1 above)
Comment (moozzyk) [reply test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs]: VS lost the setting again...

**Comment by davidfowl** [comment]
@moozzyk does the error get logged?

**Comment by moozzyk** [comment]
Yes, the error is logged.

**Pull request merged**